django/db/backends/mysql/features.py

```diff
@@ -30,6 +30,7 @@ class DatabaseFeatures(BaseDatabaseFeatures):
     supports_select_intersection = False
     supports_select_difference = False
     supports_slicing_ordering_in_compound = True
+    strindex_matches_case_insensitively = True
```

Comment:
Yes it's controlled by collation, but collation can be applied not only to databases, but also tables, columns, and even values. Django-MySQL has a 'case sensitive exact' lookup to allow you to pass a binary-collated value which forces the other side to its binary collation before comparison ([implementation](https://github.com/adamchainz/django-mysql/blob/master/django_mysql/models/lookups.py#L17)). So yeah there shouldn't be a database feature.